sandbox-backend/src/main/java/com/exadel/sandbox/team5/dao/DiscountDAO.java

```diff
@@ -27,14 +29,17 @@
                             AND (:tags is null or t.tagName = :tags)
                             AND (:country is null or c.name = :country)
                             AND (:cities is null or s.name = :cities)
-            GROUP BY d.id
+            GROUP BY d.id #{#pageable}
                 HAVING rate>=(:rate);
-            """, nativeQuery = true)
-    List<Discount> findDiscountsByCriteria(@Param("name") String searchText,
+            """,
+            countQuery = "SELECT count(*) FROM discount"
```

Comment:
The count query also should use the same conditions as the search query.

Reply:
Done